test/unit/db/repos/accounts.js

```diff
@@ -549,6 +549,41 @@ describe('db', () => {
 						});
 				});
 			});
+
+			it('should execute all queries in one database transaction', done => {
+				const account = accountFixtures.Account();
+
+				db.$config.options.query = function(event) {
+					if (!(event.ctx && event.ctx.isTX && event.ctx.txLevel === 0)) {
```

Comment:
Also add a check for transaction label `Accounts:setAccountAndGet`

Reply:
In this case the label will be `db:accounts:upsert`